### ISSUE TITLE
exported: comment starting with "Deprecated:” should be ignored when checking if a comment is present

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1231]
```go

// Foo does the magic
func Foo() {}

// Bar is blah blah
//
// Deprecated: use Foo
func Bar() {}

//nolint:whatever
func Baz() {}

// Deprecated: is deprecated
func Qux() {}

func Quux() {}
```

The first two are valid

The third one is fixed by a previous PR:
- #1209

And is now reported as: "exported func Baz should have comment or be unexported"

The last one is reported as: "exported func Quux should have comment or be unexported"

The issue is with `Qux`, the error reported is "comment should be like Qux ...", but in fact here it should be "exported function should have a comment", the very same way `Baz` is reported

**Comment by ccoVeille** [comment]
Reminder:

> nit: in the next PR please add a comment explaining that ` .Text`  removes directives from the comment block
_Originally posted by @chavacava in https://github.com/mgechev/revive/pull/1209#discussion_r1930024105_

**Comment by chavacava** [comment]
@ccoVeille , the proposal LGTM.
